geth/peers/topicpool.go

```diff
@@ -0,0 +1,234 @@
+package peers
+
+import (
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common/mclock"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/p2p/discover"
+	"github.com/ethereum/go-ethereum/p2p/discv5"
+	"github.com/status-im/status-go/geth/params"
+)
+
+// NewTopicPool returns instance of TopicPool
+func NewTopicPool(topic discv5.Topic, limits params.Limits, slowSync, fastSync time.Duration) *TopicPool {
+	return &TopicPool{
+		topic:    topic,
+		limits:   limits,
+		slowSync: slowSync,
+		fastSync: fastSync,
+		peers:    map[discv5.NodeID]*peerInfo{},
+	}
+}
+
+// TopicPool manages peers for topic.
+type TopicPool struct {
+	topic    discv5.Topic
+	limits   params.Limits
+	slowSync time.Duration
+	fastSync time.Duration
+
+	quit    chan struct{}
+	running int32
```

Comment:
I like to use ints this way because we can rely on default value - 0. and with atomic.Value i will have to Set(false) in the constructor and maybe in tests.

additionally, most of the codebases are using ints for this purpose, just grepped go-ethereum and there are plenty of examples

Reply:
One of the parts in `go-ethereum` I'm no friend of. But it's okay, let's keep it.